### ISSUE TITLE
[Bug]: @babel/traverse scope set the binding references

### ISSUE THREAD

**Issue by xukaijie111** [babel/babel#14068]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

Programmatic API (`babel.transform`, `babel.parse`)

### Input code

```
let a = 5;
a  = 6
```

### Configuration file name

babel.config.js

### Current and expected behavior

when use babel/traverse the code ast .the scope has binding 'a',but bind.references is 0。the variable when is only readed is not referenced to the binding ?

### Environment

 System:
    OS: macOS 10.15.5
  Binaries:
    Node: 16.6.1 - ~/.nvm/versions/node/v16.6.1/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 7.20.3 - ~/.nvm/versions/node/v16.6.1/bin/npm

**Comment by nicolo-ribaudo** [comment]
If I recall correctly, we track reads in `references` and writes in `constantViolations`.

**Comment by JLHwung** [comment]
Good catch. I agree with OP. The LHS `a` in `a = 1` should be a referenced path, too.

Note that in the spec `a` is an IdentifierRefernece descended from [AssignmentPattern](https://tc39.es/ecma262/#prod-AssignmentPattern). If I read correctly, if an Identifier is [referenced](https://github.com/babel/babel/blob/main/packages/babel-types/src/validators/isReferenced.ts) in Babel's sense, it will be an IdentifierRefernece in the spec AND currently it can not descend from AssignmentPattern directly (Initializer and computed key is not considered directly). we should at least document the behaviour in that helper.
